src/core/qgspallabeling.cpp

```diff
@@ -3944,7 +3944,7 @@ QgsGeometry* QgsPalLabeling::prepareGeometry( const QgsGeometry* geometry, QgsRe
     clonedGeometry.reset( geom );
   }
 
-  if ( clipGeometry && !clipGeometry->contains( geom ) )
+  if ( clipGeometry && !clipGeometry->boundingBox().contains( geom->boundingBox() ) )
```

Comment:
But this [previous line](https://github.com/qgis/QGIS/pull/2852/files#diff-a3ce81e20eab5de8d69fffffff63d41fL3834) covers it? no?